### PR TITLE
[LC-386] remove group_id and add LOAD_PEERS_FROM_IISS option

### PR DESCRIPTION
**Pull request by windies21**
this PR pending by (https://github.com/icon-project/loopchain/pull/224).

* remove group_id
* add LOAD_PEERS_FROM_IISS option.

### CODE REVIEW

File: loopchain/baseservice/peer_manager.py [hunk 1]
```diff
@@ -25,78 +26,41 @@
...
+        self.peer_order_list: dict = {}  # { order:peer_id }
 
     def serialize(self) -> dict:
         peer_info_list_serialized = dict()
-        for group_id, peer_info_group in self.__peer_info_list.items():
-            group_serialized = dict()
-            for peer_id, peer_info in peer_info_group.items():
-                group_serialized[peer_id] = peer_info.serialize()
-            peer_info_list_serialized[group_id] = group_serialized
+        for peer_id, peer_info in self.peer_info_list.items():
```
Comment (zsaladin) [inline loopchain/baseservice/peer_manager.py]:
```python
peer_info_list_serialized = { peer_id: peer_info.serialize() 
                              for peer_id, peer_info in peer_info_list.items() }
```
pythonic?

File: loopchain/baseservice/peer_manager.py [hunk 2]
```diff
@@ -25,78 +26,41 @@
...
 
     @staticmethod
     def deserialize(peer_list_data_serialized: dict) -> 'PeerListData':
         peer_info_list = dict()
-        for group_id, peer_info_group_serialized in peer_list_data_serialized['peer_info_list'].items():
-            group = dict()
-            for peer_id, peer_info_serialized in peer_info_group_serialized.items():
-                group[peer_id] = PeerInfo.deserialize(peer_info_serialized)
-            peer_info_list[group_id] = group
+        for peer_id, peer_info_serialized in peer_list_data_serialized['peer_info_list'].items():
```
Comment (zsaladin) [inline loopchain/baseservice/peer_manager.py]:
```python
peer_info_list = { peer_id: PeerInfo.deserialize(peer_info_serialized)
                   for peer_id, peer_info_serialized in peer_list_data_serialized['peer_info_list'].items() }
```
pythonic?

File: loopchain/baseservice/peer_manager.py [hunk 3]
```diff
@@ -25,78 +26,41 @@
...
+            peer_info_list[peer_id] = PeerInfo.deserialize(peer_info_serialized)
 
         peer_order_list = dict()
-        for group_id, order_list in peer_list_data_serialized['peer_order_list'].items():
-            converted_order_list = dict()
-            for order, peer_id in order_list.items():
-                converted_order_list[int(order)] = peer_id
-            peer_order_list[group_id] = converted_order_list
+        for order, peer_id in peer_list_data_serialized['peer_order_list'].items():
```
Comment (zsaladin) [inline loopchain/baseservice/peer_manager.py]:
```python
peer_order_list = { int(order): peer_id 
                    for order, peer_id in peer_list_data_serialized['peer_order_list'].items() }
```
pythonic?

File: loopchain/baseservice/peer_manager.py [hunk 4]
```diff
@@ -116,46 +80,54 @@ def __init__(self, channel_name):
...
-        self.__init_peer_group(conf.ALL_GROUP_ID)
         # lock object for if add new peer don't have order that must locking
         self.__add_peer_lock: threading.Lock = threading.Lock()
 
         self.__leader_complain_count = 0
         self.__highest_block_height = -1    # for RS heartbeat
 
-    @property
-    def peer_object_list(self) -> dict:
+    def peer_ids_hash(self):
```
Comment (zsaladin) [inline loopchain/baseservice/peer_manager.py]: Is it replaced with rep hash?

File: loopchain/baseservice/peer_manager.py [hunk 5]
```diff
@@ -167,27 +139,12 @@ def set_peer_list(self, peer_list_data: PeerListData):
         self.peer_list_data = peer_list_data
 
     def get_peer_by_target(self, peer_target):
-        for group_id in self.peer_list.keys():
-            for peer_id in self.peer_list[group_id]:
-                peer_each = self.peer_list[group_id][peer_id]
-                if peer_each.target == peer_target:
-                    return peer_each
+        for peer_id in self.peer_list:
```
Comment (zsaladin) [inline loopchain/baseservice/peer_manager.py]:
```python
return next((peer for peer in self.peer_list.values() if peer.target == peer_target), None)
```
pythonic?

File: loopchain/rest_server/rest_server_rs.py [hunk 6]
```diff
@@ -84,7 +84,7 @@ def set_resource(self):
     def get_peer_list(self, channel):
         return self.__stub_to_rs_service.call(
             "GetPeerList",
-            loopchain_pb2.CommonRequest(request="", group_id=conf.ALL_GROUP_ID, channel=channel))
+            loopchain_pb2.CommonRequest(request="", group_id="all_group_id", channel=channel))
```
Comment (zsaladin) [inline loopchain/rest_server/rest_server_rs.py]: This group ID is not deleted.

File: loopchain/rest_server/rest_server_rs.py [hunk 7]
```diff
@@ -209,21 +209,21 @@ class Peer(HTTPMethodView):
 
             if peer_manager.peer_list:
                 async_futures: List[grpc.Future] = []
-                for peer_id in peer_manager.peer_list[conf.ALL_GROUP_ID]:
-                    async_future = ServerComponents().get_peer_status_async(peer_id, conf.ALL_GROUP_ID, channel)
+                for peer_id in peer_manager.peer_list:
+                    async_future = ServerComponents().get_peer_status_async(peer_id, "all_group_id", channel)
```
Comment (zsaladin) [inline loopchain/rest_server/rest_server_rs.py]: This group ID is not deleted.

File: loopchain/peer/peer_service.py [hunk 8]
```diff
@@ -169,7 +156,7 @@ def __get_channel_infos(self):
                     message=loopchain_pb2.GetChannelInfosRequest(
                         peer_id=self.__peer_id,
                         peer_target=self.__peer_target,
-                        group_id=self.group_id),
+                        group_id=self.__peer_id),
```
Comment (zsaladin) [inline loopchain/peer/peer_service.py]: This group_id should be deleted.

File: loopchain/peer/peer_inner_service.py [hunk 9]
```diff
@@ -40,7 +40,7 @@ def __init__(self, peer_service: 'PeerService'):
 
         return \
             self._peer_service.peer_port, self._peer_service.peer_target, self._peer_service.rest_target, \
-            self._peer_service.radio_station_target, self._peer_service.peer_id, self._peer_service.group_id, \
+            self._peer_service.radio_station_target, self._peer_service.peer_id, "", \
```
Comment (zsaladin) [inline loopchain/peer/peer_inner_service.py]: This group_id should be deleted.

File: loopchain/channel/channel_service.py [hunk 10]
```diff
@@ -472,7 +504,7 @@ def connect_to_radio_station(self, is_reconnect=False):
                 peer_object=b'',
                 peer_id=ChannelProperty().peer_id,
                 peer_target=ChannelProperty().peer_target,
-                group_id=ChannelProperty().group_id),
+                group_id=ChannelProperty().peer_id),
```
Comment (zsaladin) [inline loopchain/channel/channel_service.py]: This group_id should be deleted.

File: loopchain/channel/channel_service.py [hunk 11]
```diff
@@ -532,7 +564,7 @@ def __subscribe_to_peer_list(self):
                 loopchain_pb2.PeerRequest(
                     channel=ChannelProperty().name,
                     peer_target=ChannelProperty().peer_target, peer_type=peer_type,
-                    peer_id=ChannelProperty().peer_id, group_id=ChannelProperty().group_id,
+                    peer_id=ChannelProperty().peer_id, group_id=ChannelProperty().peer_id,
```
Comment (zsaladin) [inline loopchain/channel/channel_service.py]: This group_id should be deleted.

File: loopchain/channel/channel_service.py [hunk 12]
```diff
@@ -509,7 +541,7 @@ def __subscribe_to_peer_list(self):
         peer_request = loopchain_pb2.PeerRequest(
             channel=ChannelProperty().name,
             peer_target=ChannelProperty().peer_target,
-            peer_id=ChannelProperty().peer_id, group_id=ChannelProperty().group_id,
+            peer_id=ChannelProperty().peer_id, group_id=ChannelProperty().peer_id,
```
Comment (zsaladin) [inline loopchain/channel/channel_service.py]: This group_id should be deleted.

**Comment by yakkle** [comment]
Please update title with summary of this feature.

File: loopchain/baseservice/peer_manager.py (hunk 1 above)
Comment (laurenceyoon) [reply loopchain/baseservice/peer_manager.py]: Very pythonic :)

File: loopchain/baseservice/peer_manager.py (hunk 4 above)
Comment (laurenceyoon) [reply loopchain/baseservice/peer_manager.py]: Replacing repHash will be updated at the next pull request.

File: loopchain/rest_server/rest_server_rs.py (hunk 6 above)
Comment (laurenceyoon) [reply loopchain/rest_server/rest_server_rs.py]: `group_id` is partially removed here. Group ids related to grpc proto will be updated at the next pull request.

File: loopchain/rest_server/rest_server_rs.py (hunk 7 above)
Comment (laurenceyoon) [reply loopchain/rest_server/rest_server_rs.py]: `group_id` is partially removed here. Group ids related to grpc proto will be updated at the next pull request.

File: loopchain/peer/peer_service.py (hunk 8 above)
Comment (laurenceyoon) [reply loopchain/peer/peer_service.py]: `group_id` is partially removed here. Group ids related to grpc proto will be updated at the next pull request.

File: loopchain/channel/channel_service.py (hunk 10 above)
Comment (laurenceyoon) [reply loopchain/channel/channel_service.py]: `group_id` is partially removed here. Group ids related to grpc proto will be updated at the next pull request.

File: loopchain/channel/channel_service.py (hunk 11 above)
Comment (laurenceyoon) [reply loopchain/channel/channel_service.py]: `group_id` is partially removed here. Group ids related to grpc proto will be updated at the next pull request.

File: loopchain/channel/channel_service.py (hunk 12 above)
Comment (laurenceyoon) [reply loopchain/channel/channel_service.py]: `group_id` is partially removed here. Group ids related to grpc proto will be updated at the next pull request.

**Pull request merged**